src/com/amazon/ion/impl/bin/utf8/Utf8StringEncoderPool.java

```diff
@@ -0,0 +1,60 @@
+package com.amazon.ion.impl.bin.utf8;
+
+import java.util.concurrent.ArrayBlockingQueue;
+
+/**
+ * A thread-safe shared pool of {@link Utf8StringEncoder}s that can be used for UTF8 encoding and decoding.
+ */
+public class Utf8StringEncoderPool {
```

Comment:
Why not make this generic? Nothing about this class seems to be specific to the UTF-8 encoder use case (assuming that you drop the singleton pattern, but even then you can have a singleton UTF-8 encoder pool that is an instantiation of a generic pool).

Reply:
This pattern may be useful for more cases than just UTF-8 string encoding- do we have any other object pools in ion-java?

Reply:
> This pattern may be useful for more cases than just UTF-8 string encoding- do we have any other object pools in ion-java?

There's the `PooledBlockAllocatorProvider`, but I believe that's it. (Depending on your perspective, the [`RecyclingStack`](https://github.com/amzn/ion-java/blob/0eb62ba92633f03e9f3175bb77f1b3214fcc1cdd/src/com/amazon/ion/impl/bin/IonRawBinaryWriter.java#L486) might qualify?) At any rate, I plan to tackle #370 next, at which point we'll definitely have an opportunity for reuse.

> Why not make this generic? Nothing about this class seems to be specific to the UTF-8 encoder use case (assuming that you drop the singleton pattern, but even then you can have a singleton UTF-8 encoder pool that is an instantiation of a generic pool).

Agreed. I'll do this as part of the PR for #370 if you don't mind.